adapters/repos/db/lsmkv/store.go

```diff
@@ -446,19 +447,28 @@ func (s *Store) ReplaceBuckets(ctx context.Context, bucketName, replacementBucke
 	currReplacementBucketDir := replacementBucket.dir
 	newReplacementBucketDir := currBucketDir
 
+	if err := bucket.Shutdown(ctx); err != nil {
+		return errors.Wrapf(err, "failed shutting down bucket old '%s'", bucketName)
+	}
+
+	replacementBucket.flushLock.Lock()
 	if err := os.Rename(currBucketDir, newBucketDir); err != nil {
+		replacementBucket.disk.maintenanceLock.Unlock()
+		replacementBucket.flushLock.Unlock()
+		s.bucketAccessLock.Unlock()
 		return errors.Wrapf(err, "failed moving orig bucket dir '%s'", currBucketDir)
 	}
 	if err := os.Rename(currReplacementBucketDir, newReplacementBucketDir); err != nil {
+		replacementBucket.disk.maintenanceLock.Unlock()
```

Comment:
you are unlocking `disk.maintenanceLock` in case of errors in line 456 and here, but I can't see that you would do this in case of success, you only unlock the `flushLock` in line 467 but not `disk.maintenanceLock`, is that intended?